### DISCUSSION TITLE
Use a custom sound for the "remote ringing" feature

### DISCUSSION THREAD

**Discussion by DodoLeDev** [andyholmes/valent#542] [Feature Request]
Hi everyone, first and foremost, thank you so much Andy Holmes for this program!
I am now using it for several months, and I love it… _when it succeeds in connecting to my phone!_ (I often encounter the problem that my phone can't be found on the network, but that's not the topic for now)

After deeply tweaking my sound themes on my computer, I found that Valent plays the `phone-incoming-call.oga`, from the `freedesktop` sound theme, when I ring my computer remotely.
But after changing this reference, by creating a custom sound theme with a different `phone-incoming-call.oga` file,
I discovered that Valent doesn't use it, but uses its [own file](https://github.com/andyholmes/valent/blob/main/src/plugins/findmyphone/data/ring.oga) instead, as we can see [here](https://github.com/andyholmes/valent/blob/main/src/plugins/findmyphone/valent-findmyphone-ringer.c#L113)
```c
g_object_set (ringer->playbin,
                    "uri", "resource:///plugins/findmyphone/ring.oga",
                    NULL);
```

What I would suggest, is to **use the system-wide sound reference** called `phone-incoming-call` of the current sound theme, and if none is found (_which I highly doubt_), use the Valent one.

What do you think?

**Comment by andyholmes** [comment]
I'm not sure what I want to do about this, to be honest. The reality of sound themes is that they are awful and I don't think anyone wants to fix the situation.

I avoided it entirely because the obvious thing to use is libcanberra with GSound, and I've been down that road and it required maintaining one or two fallbacks. For that to work at all in a Flatpak, Valent would have to be given permissions for the host filesystem, which is not going to happen :)

So the answer is I'm not really sure what I want to do about it, but sound themes are probably not the solution I'd choose. Maybe just a box to select a sound file, but that's not very awesome either.

**Reply by andyholmes** [reply]
Ah, this looks like what we will want: https://github.com/flatpak/xdg-desktop-portal/issues/929

**Reply by DodoLeDev** [reply]
> Maybe just a box to select a sound file

Yeah that's a good solution! :smiley:

**Reply by andyholmes** [reply]
> Ah, this looks like what we will want: https://github.com/flatpak/xdg-desktop-portal/issues/929

I think this will be the proper solution. A plain sound file is really not friendly for accessibility users.
